### PR TITLE
Fix #3600: Add complete UI for tablet devices for copyright licenses lists and texts

### LINKED ISSUE #3600: Hi-fi (Tablet) for displaying third-party dependencies and their license links
The dependency list and licenses list should be displayed side by side in large screen-size devices.

Mock link - https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/dae1f41b-9551-430e-8b61-b285d6ee050b/

### PR DESCRIPTION
**Pull request by prayutsu**
## Explanation

 Fixes #3600:
Added split mode support on tablet devices for LicenseListFragment and LicenseTextViewerFragment.

| Portrait | Landscape |
| --- | --- |
| Screenshot_1629013550 | Screenshot_1629013557|
|Screenshot_1629013803|Screenshot_1629013792|
|Screenshot_1629013904|Screenshot_1629013897|
|Screenshot_1629013880|Screenshot_1629013886|
|Screenshot_2021-08-15-13-33-21-234_org oppia android|Screenshot_2021-08-15-13-29-51-448_org oppia android|
|Screenshot_2021-08-15-13-29-54-437_org oppia android|Screenshot_2021-08-15-13-36-23-887_org oppia android|

Test cases-
Screenshot from 2021-08-16 21-43-27
(3 test cases failing in espresso that are only meant for mobile)
Screenshot from 2021-08-16 21-49-05

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by prayutsu** [comment]
@rt4914 UI is now behaving as expected, only 1 bug is left. The app is crashing if we follow these steps-
1. Select third-party dependencies.
2. Select one of the items from the list in right.
3. Rotate the tablet device.
4. Press back.
5. Now select some dependency item again.
6. Notice that app crashes with the following error - 

Screenshot from 2021-08-15 13-00-01


https://user-images.githubusercontent.com/54636525/129470921-b0f56bef-adfd-4276-ba20-e7849834c1ba.mp4

**Comment by prayutsu** [comment]
@rt4914 I will add test cases once the bug is resolved.

**Comment by rt4914** [comment]
> @rt4914 UI is now behaving as expected, only 1 bug is left. The app is crashing if we follow these steps-
> 
> 1. Select third-party dependencies.
> 2. Select one of the items from the list in right.
> 3. Rotate the tablet device.
> 4. Press back.
> 5. Now select some dependency item again.
> 6. Notice that app crashes with the following error -
> 
> Screenshot from 2021-08-15 13-00-01
> 
>  crash.mp4

@prayutsu I think instead of adding fragments to backstack just replace the fragment on right completely and on back-button click determine which fragment is currently shown  and based on that close the current fragment and replace with a new parent fragment. 

WDYT?

**Comment by prayutsu** [comment]
@rt4914 PTAL.

**Review by BenHenning — APPROVED** [review]
Thanks @prayutsu. LGTM for codeowners.

@rt4914 deferring to you to block this until the base PR is merged.

File: app/src/main/java/org/oppia/android/app/help/HelpActivity.kt
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/help/HelpActivity.kt:55]: This can be shifted to single line.

File: app/src/main/java/org/oppia/android/app/help/HelpActivityPresenter.kt
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/help/HelpActivityPresenter.kt]: What is the use of these format arguments? `"third-party dependencies list"`

File: app/src/main/java/org/oppia/android/app/help/thirdparty/LicenseTextViewerFragment.kt
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/help/thirdparty/LicenseTextViewerFragment.kt]: This change is not required

File: app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt]: The `,` at the ned needs to be removed right?

File: app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt]: You can removed this `?` too

File: app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt]: Please update this.

File: app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt]: Update here and at other places
` matches(withEffectiveVisibility(Visibility.VISIBLE))`

File: app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt]: Ditto

**Review by rt4914 — CHANGES_REQUESTED** [review]
@prayutsu Overall good. Can you add one test case for [this](https://github.com/oppia/oppia-android/pull/3680#issuecomment-899010266) case too ?

File: app/src/main/java/org/oppia/android/app/help/HelpActivity.kt
Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/help/HelpActivity.kt:55]: Actually, it is crossing the max 100 char limit when I am trying to fit it into a single line

File: app/src/main/java/org/oppia/android/app/help/HelpActivityPresenter.kt
Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/help/HelpActivityPresenter.kt]: As per your [earlier comment](https://github.com/oppia/oppia-android/pull/3671#issuecomment-899274381), I am setting the content description dynamically.

File: app/src/main/java/org/oppia/android/app/help/thirdparty/LicenseTextViewerFragment.kt
Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/help/thirdparty/LicenseTextViewerFragment.kt]: Reverted.

File: app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt
Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt]: Removed.

File: app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt
Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt]: Removed

File: app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt
Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/help/thirdparty/ThirdPartyDependencyListFragment.kt]: Updated.

File: app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt
Comment (prayutsu) [reply app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt
Comment (prayutsu) [reply app/src/sharedTest/java/org/oppia/android/app/help/HelpFragmentTest.kt]: Done.

**Comment by prayutsu** [comment]
Screenshots of new test case passing-
Screenshot from 2021-08-18 14-26-28
Screenshot from 2021-08-18 14-16-14

File: app/src/main/java/org/oppia/android/app/help/HelpActivity.kt
Comment (rt4914) [reply app/src/main/java/org/oppia/android/app/help/HelpActivity.kt:55]: Ack

File: app/src/main/java/org/oppia/android/app/help/HelpActivityPresenter.kt
Comment (rt4914) [reply app/src/main/java/org/oppia/android/app/help/HelpActivityPresenter.kt]: Please don't set it in hard-code form otherwise we won't be able to translate this to other languages. These strings should be defined in `strings.xml` file and then should be used here.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@prayutsu One change suggested in open comments related to `content-description`

File: app/src/main/java/org/oppia/android/app/help/HelpActivityPresenter.kt
Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/help/HelpActivityPresenter.kt]: Done.

**Comment by prayutsu** [comment]
PTAL @rt4914

**Review by rt4914 — APPROVED** [review]
LGTM, thanks

**Pull request merged**
